extensions/amp-form/0.1/amp-form.js

```diff
@@ -146,6 +147,16 @@ export class AmpForm {
     /** @private {?string} */
     this.state_ = null;
 
+    // Update the form non-xhr action to add `__amp_source_origin` parameter.
+    // This allows publishers to understand where the request is coming from.
+    const action = this.form_.getAttribute('action');
+    user().assert(action, 'form action is required: %s',
```

Comment:
It's in the validator, but I think we also like to do validation here for an early error indicator.